### PR TITLE
feat: support webhook validate plugin

### PR DESCRIPTION
**Pull request by AlinsRan**
### Type of change:

- [ ] Bugfix
- [x] New feature provided
- [x] Improve performance
- [ ] Backport patches

### What this PR does / why we need it:

* #1352 

* Only retain information in the debug environment, otherwise `make e2e-test` will easily cause the machine to crash.

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix-ingress-controller#community) first**

### CODE REVIEW

**Comment by tao12345666333** [comment]
Please resolve conflicts, thanks

File: test/e2e/scaffold/scaffold.go [hunk 1]
```diff
@@ -637,19 +634,6 @@ func waitExponentialBackoff(condFunc func() (bool, error)) error {
 	return wait.ExponentialBackoff(backoff, condFunc)
 }
 
-// generateWebhookCert generates signed certs of webhook and create the corresponding secret by running a script.
-func generateWebhookCert(ns string) error {
-	commandTemplate := `testdata/webhook-create-signed-cert.sh`
```
Comment (lingsamuel) [inline test/e2e/scaffold/scaffold.go:642]: It seems that the old script is not required anymore? We should delete it

**Review by lingsamuel — APPROVED** [review]

**Review by tao12345666333 — COMMENTED** [review]
Please resolve conflicts, thanks

File: test/e2e/scaffold/scaffold.go (hunk 1 above)
Comment (AlinsRan) [reply test/e2e/scaffold/scaffold.go:655]: Done

**Review by lingsamuel — APPROVED** [review]

**Review by tao12345666333 — APPROVED** [review]

**Pull request merged**